components/Currency/basic_usage.rst

```diff
@@ -0,0 +1,202 @@
+
+The **CurrencyContext** gives you the ability to quickly
+set and retrieve currency names from a given :doc:`/components/Storage/index`.
+
+.. tip::
+   You can use a custom storage, as long as it implements the :ref:`component_storage_storage-interface`.
+
+In this example let's use the default :ref:`component_storage_session-storage`.
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Currency\Context\CurrencyContext;
+   use Sylius\Component\Storage\SessionStorage;
+   use Symfony\Component\HttpFoundation\Session\Session;
+
+   $session = new Session();
+   $session->start();
+   $sessionStorage = new SessionStorage($session);
+
+   $currency = 'USD'; // the currency which will be used by default in this context
+
+   $currencyContext = new CurrencyContext($sessionStorage, $currency);
+
+   $currencyContext->getDefaultCurrency(); // returns 'USD'
+   $currencyContext->getCurrency('GBP'); // returns 'USD' as the currency given is not in storage
+   $currencyContext->setCurrency('GBP');
+   $currencyContext->getCurrency('GBP'); // returns 'GBP' for now it's available in the storage
+
+.. note::
+   This service implements the :ref:`component_currency_context_currency-context-interface`. |br|
+   For more detailed information go to `Sylius API CurrencyContext`_.
+
+.. _Sylius API CurrencyContext: http://api.sylius.org/Sylius/Component/Currency/Context/CurrencyContext.html
+
+.. caution::
+   Be aware that setting the default currency is done only once while creating the context,
+   afterwards you cannot change it.
+
+.. _component_currency_converter_currency-converter:
+
+CurrencyConverter
+-----------------
+
+The **CurrencyConverter** allows you to convert a value accordingly to the exchange rate of specified currency.
+
+.. code-block:: php
+
+   <?php
+
+   use Sylius\Component\Currency\Converter\CurrencyConverter;
+   use Sylius\Component\Currency\Model\Currency;
+   use Sylius\Component\Resource\Repository\InMemoryRepository;
+
+   $currencyRepository = new InMemoryRepository();
+
+   $currency = new Currency();
+   $currency->setCode('USD');
+   $currency->setExchangeRate(1.5);
+
+   $currencyConverter = new CurrencyConverter($currencyRepository);
+
+   $currencyConverter->convert(1000, 'USD'); // returns 1500
+
+.. note::
+   This service implements the :ref:`component_currency_converter_currency-converter-interface`. |br|
+   For more detailed information go to `Sylius API CurrencyConverter`_.
+
+.. _Sylius API CurrencyConverter: http://api.sylius.org/Sylius/Component/Currency/Converter/CurrencyConverter.html
+
+.. caution::
+   Throws :ref:`component_currency_converter_unavailable-currency-exception`.
+
+Importers
+---------
+
+Importers allow you to get up-to-date exchange rates from given provider.
+The rates are set from your chosen base currency to all other managed currencies.
+
+.. _component_currency_importer_european-central-bank-importer:
+
+EuropeanCentralBankImporter
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I think we should move this and AbstractImporter into separate section -> importers or Importing exchange rates.